twoliter/embedded/imghelper

```diff
@@ -108,6 +110,214 @@ mkfs_data_xfs() {
+
+  sbsetup_wrapup local
+}
+
+uefi_pesign() {
+  local -n shim_sign_key code_sign_key
+  local arch shims shim grubs grub mokms mokm
+  shim_sign_key="${1:?}"
+  code_sign_key="${2:?}"
+  arch="${3:?}"
+  shims=(boot*.efi)
+  shim="${shims[0]}"
+  grubs=(grub*.efi)
+  grub="${grubs[0]}"
+  mokms=(mm*.efi)
+  mokm="${mokms[0]}"
+
+  # Convert the vendor certificate to the expected format.
+  cert_table "${SBKEYS}/vendor.cer" "${SBKEYS}/vendor.obj"
+
+  # Replace the embedded vendor certificate, then sign shim with the db key.
+  if [[ "$(pesign -i "${shim}" -l)" == "No signatures found." ]]; then
+    mv "${shim}" "${shim}.unsigned"
+  else
+    pesign -i "${shim}" -o "${shim}.unsigned" -u 0 -r
+  fi
+  "${arch}-objdumpcopy" "${shim}.unsigned" \
+    --update-section ".vendor_cert=${SBKEYS}/vendor.obj"
+  pesign -i "${shim}.unsigned" -o "${shim}.signed" -s "${shim_sign_key[@]}"
+  mv "${shim}.signed" "${shim}"
+  rm "${shim}.unsigned"
+  pesign -i "${shim}" -l
+  pesigcheck -i "${shim}" -n 0 -c "${SBKEYS}/db.cer"
+
+  # Remove the original key if necessary, then sign the MOK manager as well.
+  if [[ "$(pesign -i "${mokm}" -l)" == "No signatures found." ]]; then
+    mv "${mokm}" "${mokm}.unsigned"
+  else
+    pesign -i "${mokm}" -o "${mokm}.unsigned" -u 0 -r
+  fi
+  pesign -i "${mokm}.unsigned" -o "${mokm}.signed" -s "${code_sign_key[@]}"
+  mv "${mokm}.signed" "${mokm}"
+  rm "${mokm}.unsigned"
+  pesign -i "${mokm}" -l
+  pesigcheck -i "${mokm}" -n 0 -c "${SBKEYS}/vendor.cer"
+
+  # Remove the original vendor key if necessary, replace the embedded gpg
+  # public key, then sign grub with the vendor key.
+  if [[ "$(pesign -i "${grub}" -l)" == "No signatures found." ]]; then
+    mv "${grub}" "${grub}.unsigned"
+  else
+    pesign -i "${grub}" -o "${grub}.unsigned" -u 0 -r
+  fi
+  "${arch}-objdumpcopy" "${grub}.unsigned" \
+    --file-alignment 4096 \
+    --update-section ".pubkey=${SBKEYS}/config-sign.pubkey"
+  pesign -i "${grub}.unsigned" -o "${grub}.signed" -s "${code_sign_key[@]}"
+  mv "${grub}.signed" "${grub}"
+  rm "${grub}.unsigned"
+  pesign -i "${grub}" -l
+  pesigcheck -i "${grub}" -n 0 -c "${SBKEYS}/vendor.cer"
+}
+
+uefi_dummysign() {
+  local arch shims shim grubs grub
+  arch="${1:?}"
+  shims=(boot*.efi)
+  shim="${shims[0]}"
+  grubs=(grub*.efi)
+  grub="${grubs[0]}"
+
+  # Generate a zero-sized certificate in the expected format.
+  cert_table /dev/null "${SBKEYS}/vendor.obj"
+
+  # Replace the embedded vendor certificate with the zero-sized one, which
+  # shim will ignore when Secure Boot is disabled.
+  "${arch}-objdumpcopy" "${shim}" \
+    --update-section ".vendor_cert=${SBKEYS}/vendor.obj"
+
+  # Remove the embedded gpg public key to disable GRUB's signature checks.
+  "${arch}-objdumpcopy" "${grub}" \
+    --file-alignment 4096 \
+    --remove-section ".pubkey"
+}
+
+uefi_copycert() {
+  local efi_image
+  efi_image="${1:?}"
+  mcopy -i "${efi_image}" "${SBKEYS}"/db.{crt,cer} ::/EFI/BOOT
+}
```

Comment:
This feels a tiny bit silly, but by moving it out of `rpm2img` we no longer need **SBKEYS** outside of `imghelper`.